routes/logger.go

```diff
@@ -0,0 +1,69 @@
+package routes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"time"
+
+	"github.com/gocardless/draupnir/routes/chain"
+	"github.com/prometheus/common/log"
+)
+
+type key int
+
+const loggerKey key = 1
+
+func NewRequestLogger(logger log.Logger) chain.Middleware {
+	return func(next chain.Handler) chain.Handler {
+		return func(w http.ResponseWriter, r *http.Request) error {
+			// To capture the response, we replace the response writer with a response
+			// recorder.
+			recorder := httptest.NewRecorder()
+
+			scopedLogger := logger.With("http_request", r)
```

Comment:
I don't know if we should dump the http_request like this when we already extract the path, params etc. I feel like we've already presented all our info in the manner we'd need and can get away with not producing the messy dump of the http_request object.

Reply:
Going to leave this in for now - we can always remove it later if it's too noisy.